src/gdb_main.c

```diff
@@ -223,28 +234,36 @@ int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t
 	/* Optional GDB packet support */
 	case 'p': { /* Read single register */
 		ERROR_IF_NO_TARGET();
-		uint32_t reg;
-		sscanf(pbuf, "p%" SCNx32, &reg);
-		uint8_t val[8];
-		size_t s = target_reg_read(cur_target, reg, val, sizeof(val));
-		if (s > 0)
-			gdb_putpacket(hexify(pbuf, val, s), s * 2U);
-		else
-			gdb_putpacketz("EFF");
+		if (cur_target->reg_read) {
+			uint32_t reg;
+			sscanf(pbuf, "p%" SCNx32, &reg);
+			uint8_t val[8];
+			size_t s = target_reg_read(cur_target, reg, val, sizeof(val));
+			if (s > 0)
+				gdb_putpacket(hexify(pbuf, val, s), s * 2U);
+			else
+				gdb_putpacketz("EFF");
+		} else {
+			gdb_putpacketz("00");
+		}
 		break;
 	}
 	case 'P': { /* Write single register */
 		ERROR_IF_NO_TARGET();
-		uint32_t reg;
-		int n;
-		sscanf(pbuf, "P%" SCNx32 "=%n", &reg, &n);
-		// TODO: FIXME, VLAs considered harmful.
-		uint8_t val[strlen(&pbuf[n]) / 2U];
-		unhexify(val, pbuf + n, sizeof(val));
-		if (target_reg_write(cur_target, reg, val, sizeof(val)) > 0)
-			gdb_putpacketz("OK");
-		else
-			gdb_putpacketz("EFF");
+		if (cur_target->reg_write) {
+			uint32_t reg;
+			int n;
+			sscanf(pbuf, "P%" SCNx32 "=%n", &reg, &n);
+			// TODO: FIXME, VLAs considered harmful.
+			uint8_t val[strlen(&pbuf[n]) / 2U];
+			unhexify(val, pbuf + n, sizeof(val));
+			if (target_reg_write(cur_target, reg, val, sizeof(val)) > 0)
+				gdb_putpacketz("OK");
+			else
+				gdb_putpacketz("EFF");
+		} else {
+			gdb_putpacketz("00");
```

Comment:
This [suffers the same response problem](https://docs.jade.fyi/gnu/gdb/gdb.html#index-P-packet) and we're going to suggest the same thing here with replying with `gdb_putpacketz("OK")` and pretending the write worked